sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/aggregate/functions.scala

```diff
@@ -1135,7 +992,76 @@ abstract class CentralMomentAgg(child: Expression) extends ImperativeAggregate w
       moments(4) = buffer.getDouble(fourthMomentOffset)
     }
 
-    getStatistic(n, mean, moments)
+    if (n == 0.0) null
+    else if (n == 1.0) 0.0
```

Comment:
I don't believe we want this behavior, since these edge cases should be handled in the `getStatistic` implementation. If you see [previous PR](https://github.com/apache/spark/pull/9003) we established that `Skewness` and `Kurtosis` should yield `Double.NaN` when `n == 1.0` but other functions like `VariancePop` should yield 0.0.